### DISCUSSION TITLE
Is there a way to synchronize books on different foliate clients

### DISCUSSION THREAD

**Discussion by svnwell** [johnfactotum/foliate#824] [Ideas]
I like foliate very much. I want to synchronize foliate reading progress information between work computer and home computer. Is there any way to do that？

**Comment by archisman-panigrahi** [comment]
I found a method. I am using MEGAsync to sync my reading progress. But you can modify the method appropriately to use another sync client. All my books have the same directory structure across different computers (including same partition mount points).

I created a directory `foliate_progress` in MEGA.

Then, I moved `~/.local/share/com.github.johnfactotum.Foliate` to `/path/to/MEGAsync/foliate_progress`.

```
mv ~/.local/share/com.github.johnfactotum.Foliate  /path/to/MEGAsync/foliate_progress/
```

I ran the MEGAsync client to sync it.

Finally I symlinked this folder back to its original location.
```
ln -s  /path/to/MEGAsync/foliate_progress/com.github.johnfactotum.Foliate /home/username/.local/share/com.github.johnfactotum.Foliate
```

Do this in all your computers. Afterwards, run MEGA to sync the progress.

**Reply by archisman-panigrahi** [reply]
@johnfactotum I don't know the mechanism of Foliate's last read page. Is it necessary to have the same directory structure across different computers? If I have the identical epub but at a different location in a different computer, will that work?

**Reply by johnfactotum** [reply]
Yes, it will work, though in that case you should exclude `library/uri-store.json` from syncing if you want to open books from the library view.

**Reply by rie39414** [reply]
thank you both: this was very helpful to me, now i use foliate instead of calibre :)

for those who use *filen.io* cloud, you have to create a remote folder and synchronize it **bidirectionally** with the local folder `/home/YOURUSERNAME/.var/app/com.github.johnfactotum.Foliate`, then in the synchronization settings **exclude** with the ignore file function the file `uri-store.json` in that kind of notepad that appears.
This will allow you to open the same book from the foliate library from multiple PCs and find the same highilits.

heartfelt thanks to @johnfactotum for this awesome program

update: _foliate_ in the library don´t recongnize opened books at restart. i have to use _calibre_ to open books with foliate app (luckly highlights are still sync)
